packages/langium/src/parser/token-builder.ts

```diff
@@ -42,6 +64,16 @@ export class DefaultTokenBuilder implements TokenBuilder {
         return tokens;
     }
 
+    popLexingReport(_text: string): LexingReport {
```

Comment:
I think a name like `finalize` or `finalizeLexing` would probably be more user-friendly

Reply:
Ah, the difficulty of good naming. I was debating over this and I previously had `flushLexingReport` to indicate that afterwards the state is reset. My issue with `finalizeX` or `endX` or `stopX` was the asymmetry because we do not have the opposing `startX` or `beginX`. In the end, I chose `popLexingReport` as this is what was used in the indendation-based token provider with the `popRemainingDedents`. But as usual, no hard opinions.

Reply:
Yeah that makes sense, but `pop` also implies a corresponding `push` function. I like the `flush` variant more 👍